BLSQ/iaso
# IA-2492:  display default permission error or back-end error on 403

While having a permission error on save, we should display better error 

Related JIRA tickets : IA-2492
## Self proofreading checklist


## Changes

Adapt `useSnackMutation` to display custom error message while having a 403

## How to test

Try to save something on iaso while you don't the permission to do it.

I the backend provide a messge you should see it in the snackbar, otherwise you should see a default message

## Print screen / video



https://github.com/BLSQ/iaso/assets/12494624/4a8624a8-98d3-4a0c-8659-2d21e91b768c
